packages/web/src/hooks/use-app-status.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * Module dependencies.
+ */
+
+import { useCallback, useEffect, useState } from 'react';
+import { useContracts } from 'src/context/contracts';
+import fromUnixTime from 'date-fns/fromUnixTime';
+import isAfter from 'date-fns/isAfter';
+import isBefore from 'date-fns/isBefore';
+
+/**
+ * Current date.
+ */
+
+// TODO: Should come from the server.
+const currentDate = new Date().getTime();
+
+/**
+ * App state.
+ */
+
+export const appState = {
+  comingSoon: 'SOON',
+  countdown: 'COUNTDOWN',
+  sale: 'SALE',
+  vesting: 'VESTING'
+} as const;
+
+/**
+ * `Status` type.
+ */
+
+type Status = typeof appState[keyof typeof appState];
+
+/**
+ * Normalize unix time.
+ */
+
+function normalizeTime(unixTime) {
+  return fromUnixTime(unixTime.toString());
+}
+
+/**
+ * Export `useAppStatus` hook.
+ */
+
+export function useAppStatus() {
+  const [status, setStatus] = useState<Status | undefined>();
+  const contracts = useContracts();
+  const getStatus = useCallback(async () => {
+    if (!contracts?.sale1 || !contracts.vesting) {
+      return;
+    }
+
+    try {
+      const saleStart = await contracts.sale1.start();
+      const saleEnd = await contracts.sale1.end();
+      const vestingStart = await contracts.vesting.startTime();
+      const saleStartDate = normalizeTime(saleStart);
+      const saleEndDate = normalizeTime(saleEnd);
+      const vestingStartDate = normalizeTime(vestingStart);
+
+      if (isAfter(currentDate, vestingStartDate)) {
+        setStatus(appState.vesting);
```

Comment:
since the state is typed, you don't need to do this `appState.vesting` => `setStatus('VESTING')`
The same for others